### PR TITLE
Switch from Travis CI to CircleCI

### LINKED ISSUE #120: Move to CircleCI from Travis
[Travis contains a `pip` cache](https://docs.travis-ci.com/user/caching/). This would really help speed up builds as Miniconda doesn't have to be installed, and Packages just have to be cached.

The downsides are managing the cache effectively.

Is this worth considering?

### PR DESCRIPTION
**Pull request by gforsyth**
Jobs set to run in parallel and cache their respective conda environments.

Resolves #120 

Someone with admin rights will have to enable to circle hooks, and then I can rebase to trigger the job to make sure everything is working.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            source activate ${ENV_NAME}
+            pip install -e .[tests]
+      - run:
+          command: |
+            /home/circleci/miniconda/envs/py36-sparse-test/bin/pytest --flake8
+      - run:
+          command: |
+            /home/circleci/miniconda/envs/py36-sparse-test/bin/codecov
+
+  build_27:
```
Comment (hameerabbasi) [inline .circleci/config.yml]: There seems to be a lot of repetition between the two builds. Is there some way to bring that down?

File: ci/environment-2.7.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+dependencies:
+  - python=2.7
+  - pytest
+  - numpy
+  - scipy
+  - pytest-cov
+  - nomkl
+  - numba
+  - pip:
+      - pytest-flake8
```
Comment (hameerabbasi) [inline ci/environment-2.7.yml]: Can we move this to conda-forge?

File: ci/environment-3.6.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+dependencies:
+  - python=3.6
+  - pytest
+  - numpy
+  - scipy
+  - pytest-cov
+  - nomkl
+  - numba
+  - pip:
+      - pytest-flake8
```
Comment (hameerabbasi) [inline ci/environment-3.6.yml]: Same as above.

**Comment by gforsyth** [comment]
Added a badge to the README and switched the environment files to use conda-forge instead of defaults.  There might be some way to avoid the build repetition, but I'm not familiar enough with circle 2 to know where (and to not bork up the cache restore).

**Comment by hameerabbasi** [comment]
Hmm. Could you make a minor change? CircleCI was set to not build forked PRs.

**Comment by hameerabbasi** [comment]
I turned off Travis builds completely, and made CircleCI required instead. Can you re-trigger with some change?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          paths:
+            - "/home/circleci/miniconda"
+      - run:
+          command: |
+            export PATH="$HOME/miniconda/bin:$PATH"
+            source activate ${ENV_NAME}
+            pip install -e .[tests]
+      - run:
+          command: |
+            /home/circleci/miniconda/envs/py36-sparse-test/bin/pytest --flake8
```
Comment (hameerabbasi) [inline .circleci/config.yml]: Can you cd into the project directory and run `pytest`? It'll automatically pick up `pytest.ini` that way. You also won't need `--flake8`, it's already in `pytest.ini`.

This also makes the coverage not generate.

**Comment by gforsyth** [comment]
Ok, should be all set now.  Had to properly set path for the `codecov` call because apparently `codecov` is calling the `coverage` entrypoint with `subprocess` (or something similarly strange)

**Comment by hameerabbasi** [comment]
Thanks, @gforsyth! This is in.

**Pull request merged**